cpp/src/arrow/io/compressed.cc

```diff
@@ -354,16 +355,19 @@ class CompressedInputStream::Impl {
   // Try to feed more data into the decompressed_ buffer.
   Status RefillDecompressed(bool* has_data) {
     // First try to read data from the decompressor
-    if (compressed_ && compressed_->size() != 0) {
```

Comment:
I don't know whether previous check would be too strict. If `compressed_->size() == compressedPos_`, it would decompress from 0, would that be ok?

Reply:
Oh damn, this would trigger a zero-sized compression and set freshed = false...

Reply:
This branch also protect original:

```
if (!decompressed_ || decompressed_->size() == 0)
```

If `decompress_ != nullptr`, then, `compressed_` is always not nullptr. So, this branch is always hit. And during `DecompressData()`, `decompressed_` will be reset. So the original code never hit `decompressed_->size() != 0 && decompressed_->size() == decompressed_pos_`.